src/app/routes/hooks/use-on-wallet-unlock.ts

```diff
@@ -0,0 +1,10 @@
+import { useEffect } from 'react';
+
+export function useOnWalletUnlock(handler: () => void) {
+  useEffect(() => {
+    chrome.runtime.onMessage.addListener(message => {
+      if (message.method === 'unlockWallet') handler();
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+}
```

Comment:
This doesn't work. This method is fired on _all_ unlock attempts. So here, we forward all other sessions to the homepage, even if the pw is wrong. With the others, we can be fairly sure they're always going to work.

One solution would be to fire an `unlockWalletSuccess` from the bg script, when we know it's correct. Though given the changes I'm making to bg script, I'd rather we didn't do that right now.

Could we instead add a guard/redirect logic to unlock so that, on refresh, it'll redirect to home if it has a secret key in session? This isn't quite as nice a UX, but it's an edge case anyway so not a huge problem.

Reply:
Good point, I needed to test it more today. Yep, I can add the guard redirect for now and maybe revisit a better solution once your refactoring is done.